src/engine/strat_engine/mdv.rs

```diff
@@ -124,33 +143,41 @@ impl MetadataVol {
             .join(FILESYSTEM_DIR)
             .join(fs_uuid.simple().to_string())
             .with_extension("json");
-        if let Err(err) = remove_file(fs_path) {
-            if err.kind() != ErrorKind::NotFound {
-                return Err(From::from(err));
+
+        with_mount(&self.dev.devnode(), &self.mount_pt, || {
+            if let Err(err) = remove_file(fs_path) {
+                if err.kind() != ErrorKind::NotFound {
+                    return Err(From::from(err));
+                }
             }
-        }
+            Ok(())
+        })?;
 
         Ok(())
     }
 
     /// Check the current state of the MDV.
     pub fn check(&self) -> EngineResult<()> {
-        for dir_e in read_dir(self.mount_pt.join(FILESYSTEM_DIR))? {
-            let dir_e = dir_e?;
-
-            // Clean up any lingering .temp files. These should only
-            // exist if there was a crash during save_fs().
-            if dir_e.path().ends_with(".temp") {
-                match remove_file(dir_e.path()) {
-                    Err(err) => {
-                        debug!("could not remove file {:?}: {}",
-                               dir_e.path(),
-                               err.description())
+        with_mount(&self.dev.devnode(), &self.mount_pt, || {
```

Comment:
With current code, fs will be mounted every 10 seconds if nothing is happening, just to do housekeeping here. Does that seem too often?

Reply:
That `check` code is rather pointless. How about if I move the temp file removal to `setup` and we can just scrap `check`.